### ISSUE TITLE
WCAG 3 Review: Section 3.1 Text Alternatives - collection of comments

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag3#278]
It was good to be able to review the various types of examples of criteria and how they could potentially be presented in WCAG 3. Hopefully these comments are helpful. This issue contains thoughts and impressions on the Text Alternatives Guideline, How-to, Outcomes and Methods, and scoring as a whole. 

-  The Guideline short text and critical errors seem to imply that all non-text content must have an alternative or it's a failure. However, there are cases where the non-text content is decorative, so a text alternative is not needed. To the uninitiated, they read this and say – every image needs a text alternative – go write it (an incorrect interpretation). It is only by reading details in how-to examples that you learn the details that decorative images really just need to be marked properly (don’t need text developed) and alt=”” is OK. We all know this criteria is one of the most often misunderstood. Even with today's WCAG, this SC is implemented in authoring tools where there is no way to identify any images as decorative. It's important to be clear about the options more up front.  
- Some readers are more visual rather than text (reading) centric. It might be helpful to show some pictorial examples of what is non-text content and then show some good/bad examples of alternatives in their context of use. Or point to some really good articles specifically covering this. 
- Wondering if there should  be a “learn” tab (or instead include on the Design tab) that points out the visual good vs. bad examples for alternative text for the different types of non-text content. Our designers are always asking for visual examples.
- **Plan tab** The plan tab does not cover what I thought it would. Was think ing the Planning Responsibilities tab should explain that the content development team should identify who is responsible for providing the alternative text (depending on your team it may be designer, content writer, etc.), who should review it, decide if you plan to do user testing for it, when in the dev cycle you plan to test and how. You would also need to know if the tool is capable of counting the images tested along with a count of how many images pass or fail the automated test.  Since it will take some time for tools to migrate over to the new conformance model, identifying tools that support the new model may be difficult. The team instead may plan to focus on critical user paths or specific views first.
- **Confusing** This may already be planned for a future iteration of the SC... Most of the information for this guideline focuses on images (including the editor’s note), yet examples and other places in the How-to, etc. talk about additional non-text examples not covered in the ratings or other parts of the supporting material. For example, audio/video files, meaningful audio beeps/buzzes, animations, interactive visual elements like maps and data visualizations are required to have some sort of text label, and in some cases a longer description. The methods and rating/scoring also needs to take the additional types of non-text content into account - those things beyond just images.
- **Functional categories:** 
    - auditory functional categories need to be added (e.g., alternatives for beeps/buzzes given in example for a test that gives auditory indication of right/wrong answers, or the example of an audio file where one needs to provide a text name or short description of the file's content.) 
    - Since many of these functional categories are quite new to WCAG, some text explaining how alternative text is useful for all of these cognitive categories would be helpful - either in the main WCAG 3 document or in a supporting document (Outcome or How-to). For example, I’m not sure how alternative text helps in the functional category of Cognitive – Mental Health. Is this mainly with regard to understanding what interactive images are used for (e.g. icons) that can cause stress to the user? And, while this may be the case, the programmatic provision of alternative text is primarily for assistive technology. Perhaps the outcomes should be teased apart.  There are users that need to identify the usage/meaning of an interactive image in visible text. Other users need it programmatically so AT can relay it to them aurally. 
- Methods: 
    - I'm sure other methods will be added in another iteration, as additional methods are needed. e.g. for media (audio-only, video-only or multimedia),  interactive images (e.g. image maps, complex visualizations, etc.) or sounds used to indicate status (e.g. chime for success, buzzer for failure). 
    - The differences between Functional, decorative, informative, and image of text are not described in the methods. An initial clear definition (or pointer to one) would help. 
    - For scoring, not all automated test tools today count how many images there are or present the alternative text (with the context of use of the image) to make manual inspection for accuracy quicker. Additionally, manual inspection for accuracy may not be possible in very large websites, where sampling must be done. I am unsure as to how such a combination of testing could be reflected in the scoring.
    - It may be difficult for a tester of a complex application or website to quickly know what images they encounter when testing a “path” and which of those are “important”. For example, visually identifying interactive images (links, buttons, icons) and then obtaining the alt text. As a hypothetical example: There’s a number of icons in the top menu of an email application. The important tasks are identified that the user can send, receive, reply, delete, file and forward email. Those icons in the top menu have alt text, but the top level menu also has text attributes icons that do not have alt text. Are those considered “important” to the identified tasks?
